pandas/tools/pivot.py

```diff
@@ -136,8 +136,14 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
 
     table = agged
     if table.index.nlevels > 1:
-        to_unstack = [agged.index.names[i] or i
-                      for i in range(len(index), len(keys))]
+        index_names = agged.index.names[:len(index)]
```

Comment:
right now we do a best guess interpretation. We can add a private argument to ``unstack`` which will handle only indexers (it would need to be passed thru to the underlying code), so a little bit of work.

IOW

``unstack([1, 0], _level_indexers=True)`` is purely unambiguous. This is something we could use from internal code (only).